### ISSUE TITLE
Update the ActiveSupport subscribers to use monotonic clock

### ISSUE THREAD

**Issue by kaylareopelle** [newrelic/newrelic-ruby-agent#900]
I recently learned [ActiveSupport::Notifications Subscribers have a monotonic option](https://api.rubyonrails.org/classes/ActiveSupport/Notifications.html#:~:text=If%20you%20are%20concerned%20about%20accuracy%2C%20you%20can%20register%20a%20monotonic%20subscriber.). 

To align the data collected by our Rails instrumentation with our other data, I propose we update our subscribers to utilize this monotonic option.

### Related Issues
https://github.com/newrelic/newrelic-ruby-agent/issues/304 

### Additonal context:
Rails6 introduce monotonic_time, which was a [breaking change](https://github.com/rails/rails/issues/36145), but the Rails team provided [a fix](https://github.com/rails/rails/pull/36184)

**Comment by tannalynn** [comment]
I don't think this is something applicable to the ruby agent. We do not use the common way of subscription for active support notifications, we use a start/finish method system that was created specifically for us originally. 
Here is our comment describing this https://github.com/newrelic/newrelic-ruby-agent/blob/621c72558c9714ddfe61495c66761a6c8364ea08/lib/new_relic/agent/instrumentation/notifications_subscriber.rb#L50-L53
Because of our different way of subscribing, we define the start and stop time ourselves.
